src/writeData/components/WriteDataDetailsContext.tsx

```diff
@@ -101,33 +73,27 @@ const WriteDataDetailsProvider: FC = ({children}) => {
     })
   }, [variables, org?.name, update])
   useEffect(() => {
-    if (bucket?.name === variables.bucket) {
+    if (DEFAULT_BUCKET === variables.bucket) {
```

Comment:
something about these checks seem weird. Wouldn't it make more sense to listen for changes to the `variables.bucket` rather than run a bunch of useEffects just to update a specific property of the variables? I don't know, something about the useEffects sending some flags

Reply:
the control of this is inverted i think to what you expect. the component this connects to doesn't consume the provider like most of our other contexts, but instead needs to call the update function of it's parent provider with the value of variables. templates just have variables, this thing adds a structure to that open ended option. these useEffects just say "no matter what you think you're passing, the answer is DEFAULT_*"